### PR TITLE
  Tower of Hanoi(more descriptive)

### PR DESCRIPTION
**Pull request by Sakshi2517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes I have read the contributing guidelines and working according to that.

### Description
The Tower of Hanoi is the game in which it consist of three pegs or towers with n disks placed one over the other .the objective of the puzzle is to move the stack to another peg following these simple rules.
i have just included code for efficient movement of the disk so that playing this game will be ore descriptive and easier for players.
(Write your answer here.)

### Checklist

- [yes ] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [yes ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "files.associations": {
+        "BinarySearch.C": "cpp"
+    }
+}
```
Comment (HarshCasper) [inline .vscode/settings.json:5]: Remove this

File: C/other/Tower_Of_Hanoi.c [hunk 2]
```diff
@@ -1,21 +1,24 @@
 #include <stdio.h>
```
Comment (HarshCasper) [inline C/other/Tower_Of_Hanoi.c:1]: Add a description about Tower of Hanoi here

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Remove the binary file as well

File: C/other/Tower_Of_Hanoi.c (hunk 2 above)
Comment (Sakshi2517) [reply C/other/Tower_Of_Hanoi.c:1]: Tower of Hanoi consists of three pegs or towers with n disks placed one over the other.

The objective of the puzzle is to move the stack to another peg following these simple rules.

1.Only one disk can be moved at a time.
2.No disk can be placed on top of the smaller disk.
Take an example with 2 disks: Disk 1 on top of Disk 2 at peg A. The target is to move both these disks to peg B.
Move Disk 1 from peg A to peg C. Then move disk 2 from peg A to peg B and, finally, move disk 1 from peg C to peg B.

This solution takes 3 steps.

You can easily move this stack from peg B to any other peg using these 3 steps.

But what if we have 3 disks – 1,2, and 3 stacked in peg A.

To move the stack to peg B you would have to expose disk 3 and to do that disk 1 and 2 have to be moved to peg C.
So by ensuring that you do not break the rules, using the above subsets of 3 steps in the previous case, you would move disk 1 and 2 to peg C, leaving disk 3 exposed with no disk above it.
Now to solve the problem, recursively move disk 3 from peg A to peg B.

Then disk 1 from peg C to peg A. After which disk 2 can be moved above disk 3 at peg B.

The puzzle is finally completed by moving disk 1 from peg A over disk 2 and 3 at peg B.

File: .vscode/settings.json (hunk 1 above)
Comment (Sakshi2517) [reply .vscode/settings.json:5]: Sir, I am unable to remove this.would you help me please?

Comment (HarshCasper) [reply .vscode/settings.json:5]: Just remove it from the staging area and make a new commit

File: C/other/Tower_Of_Hanoi.c (hunk 2 above)
Comment (HarshCasper) [reply C/other/Tower_Of_Hanoi.c:1]: Not here 🤦 

On the Code File

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Remove that `test` file as well otherwise it would count as `spam`

**Comment by atarax665** [comment]
@Sakshi2517 Are you still working on this?

**Pull request closed without merging**